prusti-common/src/vir/to_viper.rs

```diff
@@ -113,7 +113,7 @@ impl<'v> ToViper<'v, viper::Stmt<'v>> for Stmt {
                 ast.inhale(expr.to_viper(ast), fake_position.to_viper(ast))
             }
             Stmt::Exhale(ref expr, ref pos) => {
-                assert!(!pos.is_default());
+                // assert!(!pos.is_default(), "stmt with default pos: {}", self);
```

Comment:
Do you have statements introduced via `vir!` that don't get their position reassigned to something sensible?

Reply:
yes, it seems. is there a reason `vir::Stmt::set_pos` only actually does anything for `Stmt::PackageMagicWand`?

Reply:
added `set_default_pos` calls for the statements encoding the expiration of the reference into the array as well, seems to fix it, together with adding `Stmt::Exhale` to the match in `pos`, `set_pos` in `prusti-common/src/vir/ast/stmt.rs`.

Reply:
Yeah, that looks a bit strange. I hope that this can only manifest in UI-related tests, if at all.